pandas/tests/frame/methods/test_compare.py

```diff
@@ -245,7 +245,39 @@ def test_compare_ea_and_np_dtype():
     result = df1.compare(df2, keep_shape=True)
     expected = pd.DataFrame(
         {
-            ("a", "self"): [4.0, np.nan],
+            ("a", "self"): [4.0, 4],
+            ("a", "other"): pd.Series([1, pd.NA], dtype="Int64"),
+            ("b", "self"): np.nan,
+            ("b", "other"): np.nan,
+        }
+    )
+    tm.assert_frame_equal(result, expected)
+
+
+def test_compare_with_equal_null_int64_dtypes():
+    # GH #48939
+    # two int64 nulls are considered same.
+    df1 = pd.DataFrame({"a": pd.Series([4.0, pd.NA], dtype="Int64"), "b": [1.0, 2]})
```

Comment:
The docstring of `compare` says: "Matching NaNs will not appear as a difference.", so yes, that seems the intention?
(which should be updated to be "matching NaN / NA")

Reply:
I think that wasn’t written with pd.NA in mind. So we could decide differently here, if it makes sense in the overall picture

Reply:
compare uses `pd.isna` to check missing values, and because of that it  already considers `NaN` and `NA` same. 
If compare currently says these two are equal then users would expected it to say 2 `NA`  are equal.

current behavior - 

```ipython
In [3]: df1 = pd.Series([4, np.nan])

In [4]: df2 = pd.Series([4, pd.NA], dtype='Int64')

In [5]: df1.compare(df2)
Out[5]:
Empty DataFrame
Columns: [self, other]
Index: []
```